### PR TITLE
Adds various tests for sparse data 

### LINKED ISSUE #133: Create test for K-means with sparse data

### LINKED ISSUE #138: Test the data module with sparse data structures

### LINKED ISSUE #139: Test shuffle and as_grid with sparse data
Create tests for shuffle and as_grid with sparse data

### PR DESCRIPTION
**Pull request by javicid**
# Description

Adds tests that ensure that K-Means, and the data and utils modules work correctly with sparse data structures. Various bugs had to be resolved to be able to pass these new tests.

Closes #139 
Closes #138 
Closes #133 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [ ] I have added a new test file: [E.g. `test_rf.py`]
- [x] I have added a new test case: [E.g. `RFTest.test_make_classification`]
- [x] I have tested it manually in a **local environment**.
- [ ] I have tested it manually in a **supercomputer**.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have documented the public methods of any public class according to the guide styles. 
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased my branch before trying to merge.

### CODE REVIEW

File: dislib/data/classes.py [hunk 1]
```diff
@@ -136,9 +141,17 @@ def labels(self):
...
+        if not self._sparse:
+            self._update_samples()
+        else:
+            self._update_samples_sparse()
+
         return self._samples
 
+    @property
+    def sparse(self):
+        return self._sparse
```
Comment (kafkasl) [inline dislib/data/classes.py]: it doesn't make sense to have a private attribute _sparse and then expose it as a property called sparse. Just name the attribute sparse and remove the @property

File: dislib/data/classes.py [hunk 2]
```diff
@@ -173,6 +186,13 @@ def _update_samples(self):
         for subset in self._subsets:
             self._samples = np.concatenate((self._samples, subset.samples))
 
+    def _update_samples_sparse(self):
+        self.collect()
+        self._samples = csr_matrix((0, self.n_features))
```
Comment (kafkasl) [inline dislib/data/classes.py]: I've noted that initializing an empty `_samples` causes the result to lose the information about the original `dtype`  (both in sparse and dense). I've changed it in my branch, so no need to do it here. Just letting you know

File: dislib/data/classes.py (hunk 1 above)
Comment (javicid) [reply dislib/data/classes.py]: It makes sense if you want to make the attribute immutable.

File: dislib/data/classes.py (hunk 2 above)
Comment (javicid) [reply dislib/data/classes.py]: Ok, good to know

**Pull request merged**